### PR TITLE
Provider update 3.3.0

### LINKED ISSUE #338: Upgrade to provider version 3.0.0
We will have to update the configuration to support 3.0.0 provider versions, based on this upgrade doc: https://www.terraform.io/docs/providers/google/guides/version_3_upgrade.html

### PR DESCRIPTION
**Pull request by bharathkkb**
Fixes #338

Change List for 3.3.0 upgrade:
* Remove internal references in locals to google_container_cluster.primary.region
* Default `cluster_ipv4_cidr` to null. `cluster_ipv4_cidr` is used for routes-based clusters and `ip_allocation_policy` is used for VPC-native clusters
* ${random_string.suffix.result} where networking 2.0 is used due to inability to use computed values with the new for_each implementation in subnets module

### CODE REVIEW

**Comment by morgante** [comment]
@bharathkkb What's the latest on this? Do you need anything to complete it?

**Comment by bharathkkb** [comment]
@morgante working on this currently. 3.3.0 was released today so I am making sure int tests are passing.

**Comment by bharathkkb** [comment]
@morgante actually we havent released the network module with the commits for relaxing the version constraint. If you could release it that would be awesome.

**Comment by morgante** [comment]
@bharathkkb Released - https://github.com/terraform-google-modules/terraform-google-network/blob/master/CHANGELOG.md#201---2019-12-18

**Comment by bharathkkb** [comment]
@morgante @aaron-lane  this should be ready for review

**Comment by aaron-lane** [comment]
Also, can the fix for #384 be split to a separate pull request so we can get it merged quickly and unblock other pull requests?

**Comment by bharathkkb** [comment]
@aaron-lane sg I will seperate the PRs and add upgrade guide

File: autogen/versions.tf.tmpl [hunk 1]
```diff
@@ -19,9 +19,9 @@ terraform {
 
   required_providers {
 {% if beta_cluster %}
-    google-beta = "~> 2.18"
+    google-beta = "~> 3.3.0"
```
Comment (morgante) [inline autogen/versions.tf.tmpl]: Is anything we're doing breaking compatibility with 2.18?

If not we shouldn't *force* the upgrade to 3.3.0

```suggestion
    google-beta = ">= 2.18, <4.0.0"
```

Comment (bharathkkb) [reply autogen/versions.tf.tmpl]: Done

**Comment by bharathkkb** [comment]
@aaron-lane @morgante I dont think there are any breaking changes here as I am now defaulting `cluster_ipv4_cidr` to null. Previously I had mistakenly removed this but `cluster_ipv4_cidr` is still used for routes-based clusters.

**Pull request merged**